backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/overlay/BpkOverlay.kt

```diff
@@ -0,0 +1,165 @@
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.unit.LayoutDirection
+import net.skyscanner.backpack.compose.theme.BpkTheme
+
+enum class BpkOverlayType {
+  SolidLow,
+  SolidMedium,
+  SolidHigh,
+  TopLow,
+  TopMedium,
+  TopHigh,
+  BottomLow,
+  BottomMedium,
+  BottomHigh,
+  LeftLow,
+  LeftMedium,
+  LeftHigh,
+  RightLow,
+  RightMedium,
+  RightHigh,
+  Vignette,
+}
+
+enum class GradientLevel {
+  Low,
+  Medium,
+  High,
+}
+
+enum class GradientDirection {
+  Solid,
+  Top,
+  Bottom,
+  Left,
+  Right,
+  Vignette,
+}
+
+@Composable
+fun BpkOverlayType.toGradientlevel(): GradientLevel =
+  when (this) {
+    BpkOverlayType.SolidLow,
+    BpkOverlayType.TopLow,
+    BpkOverlayType.BottomLow,
+    BpkOverlayType.LeftLow,
+    BpkOverlayType.RightLow,
+    -> GradientLevel.Low
+    BpkOverlayType.SolidMedium,
+    BpkOverlayType.TopMedium,
+    BpkOverlayType.BottomMedium,
+    BpkOverlayType.LeftMedium,
+    BpkOverlayType.RightMedium,
+    -> GradientLevel.Medium
+    BpkOverlayType.SolidHigh,
+    BpkOverlayType.TopHigh,
+    BpkOverlayType.BottomHigh,
+    BpkOverlayType.LeftHigh,
+    BpkOverlayType.RightHigh,
+    -> GradientLevel.High
+    BpkOverlayType.Vignette -> GradientLevel.High
+  }
+
+@Composable
+fun BpkOverlayType.toDirection(): GradientDirection =
+  when (this) {
+    BpkOverlayType.SolidLow,
+    BpkOverlayType.SolidMedium,
+    BpkOverlayType.SolidHigh,
+    -> GradientDirection.Solid
+    BpkOverlayType.TopLow,
+    BpkOverlayType.TopMedium,
+    BpkOverlayType.TopHigh,
+    -> GradientDirection.Top
+    BpkOverlayType.BottomLow,
+    BpkOverlayType.BottomMedium,
+    BpkOverlayType.BottomHigh,
+    -> GradientDirection.Bottom
+    BpkOverlayType.LeftLow,
+    BpkOverlayType.LeftMedium,
+    BpkOverlayType.LeftHigh,
+    -> if (LocalLayoutDirection.current != LayoutDirection.Rtl) GradientDirection.Left else GradientDirection.Right
+    BpkOverlayType.RightLow,
+    BpkOverlayType.RightMedium,
+    BpkOverlayType.RightHigh,
+    -> if (LocalLayoutDirection.current != LayoutDirection.Rtl) GradientDirection.Right else GradientDirection.Left
+    BpkOverlayType.Vignette -> GradientDirection.Vignette
+  }
+
+@Composable
+fun BpkOverlay(
+  modifier: Modifier = Modifier,
+  overlayType: BpkOverlayType = BpkOverlayType.SolidHigh,
+  foregroundContent: (@Composable BoxScope.() -> Unit)? = null,
+  content: @Composable BoxScope.() -> Unit,
+) {
+  Box(
+    modifier = modifier,
+  ) {
+    content()
+    Overlay(overlayType = overlayType)
+    if (foregroundContent != null) {
+      foregroundContent()
+    }
+  }
+}
+
+@Composable
+fun BoxScope.Overlay(
```

Comment:
Shall it be private? Doesn't seem like a public API, or it can be used by itself?

Reply:
I've made it private.